### PR TITLE
Fixes #16882: Generic method  sysctl_value

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/16882

### CODE REVIEW

File: tree/30_generic_methods/sysctl_value.cf [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+                              string => string_head("${full_inner_class_prefix_key_value}", "1000");
+
+      # we fetch on the getKey
+      "getKey"                string => canonify("get${key}");
+      "full_inner_class_prefix_variable_get"
+                              string => canonify("variable_string_from_command_sysctl_var_${getKey}_${sysctl_get_query}");
+      "inner_class_prefix_variable_get"
+                              string => string_head("${full_inner_class_prefix_variable_get}", "1000");
+
+      # we fetch on the getKey
```
Comment (Fdall) [inline tree/30_generic_methods/sysctl_value.cf]: checkKey

File: tree/30_generic_methods/sysctl_value.cf [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+                          expression => strcmp("${sysctl_var.${checkKey}}", "${value}"),
+                                  if => "${inner_class_prefix_sysctl_reload}_repaired";
+
+    any::
+      "pass3" expression => "pass2";
+      "pass2" expression => "pass1";
+      "pass1" expression => "any";
+
+  methods:
+      "disable_reporting_${class_prefix}"
```
Comment (Fdall) [inline tree/30_generic_methods/sysctl_value.cf]: should not we use full_class_prefix everywhere?

File: tests/acceptance/30_generic_methods/unsafe/sysctl_value.audit.cf [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+    # Store initial values
+      "exp1" string => execresult("${paths.sysctl} vm.swappiness > ${tmp}/backup.conf", "useshell");
+      "exp2" string => execresult("${paths.sysctl} vm.dirty_ratio >> ${tmp}/backup.conf", "useshell");
+      "exp3" string => execresult("${paths.sysctl} vm.lowmem_reserve_ratio >> ${tmp}/backup.conf", "useshell");
+      "exp4" string => execresult("${paths.sysctl} vm.oom_dump_tasks >> ${tmp}/backup.conf", "useshell");
+      "exp5" string => execresult("${paths.sysctl} vm.stat_interval >> ${tmp}/backup.conf", "useshell");
+
+
+    # REPAIR
```
Comment (Fdall) [inline tests/acceptance/30_generic_methods/unsafe/sysctl_value.audit.cf]: Can not be REPAIR if it is Audit

File: tests/acceptance/30_generic_methods/unsafe/sysctl_value.audit.cf [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+      "value[6]"     string => "999";
+      "expected[6]"  string => execresult("${paths.sysctl} -n vm.oom_dump_tasks", "useshell");
+      "filename[6]"  string => "00_rudder";
+      "option[6]"    string => "min";
+      "mode[6]"      string => "audit";
+      "status[6]"    string => "error";
+      "execute[6]"   string => "true";
+
+
+      # Audit.
```
Comment (Fdall) [inline tests/acceptance/30_generic_methods/unsafe/sysctl_value.audit.cf]: Unecesary comment

File: tests/acceptance/30_generic_methods/unsafe/sysctl_value.cf [hunk 5]
```diff
@@ -0,0 +1,179 @@
+#######################################################
+#
```
Comment (Fdall) [inline tests/acceptance/30_generic_methods/unsafe/sysctl_value.cf]: You should add a success tests after one of the repaired one

File: tests/acceptance/30_generic_methods/unsafe/sysctl_value.cf [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+# Test sysctl value setting
```
Comment (Fdall) [inline tests/acceptance/30_generic_methods/unsafe/sysctl_value.cf]: Also, what is the behaviour when using string based values and the min/max options?
It should be described in the Doc or/and in the tests

Comment (Fdall) [reply tests/acceptance/30_generic_methods/unsafe/sysctl_value.cf]: Also, is there a bitwise limit on the min/max? like there is for the maths bundles

File: tree/30_generic_methods/sysctl_value.cf (hunk 2 above)
Comment (ncharles) [reply tree/30_generic_methods/sysctl_value.cf]: YES

Comment (ncharles) [reply tree/30_generic_methods/sysctl_value.cf]: bad copy/paste

File: tests/acceptance/30_generic_methods/unsafe/sysctl_value.cf (hunk 6 above)
Comment (ncharles) [reply tests/acceptance/30_generic_methods/unsafe/sysctl_value.cf]: is numerical when possible, else allphanumerical. I'm adapting the doc

**Comment by ncharles** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
Great feedback, it allows me to find a bug ! Thank you @Fdall

File: tree/30_generic_methods/sysctl_value.cf [hunk 7]
```diff
@@ -48,6 +48,9 @@
 #  `max`: The value is the maximal value we request: the value is only changed if the current value is higher than `value`
 #  `default` (default value): The value is strictly enforced.
 #
+# Comparision is numerical is possible, else alphanumerical
```
Comment (Fdall) [inline tree/30_generic_methods/sysctl_value.cf]: Comparision of

**Comment by ncharles** [comment]
PR updated with a new commit

**Pull request merged**